From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Sat, 25 Feb 2017 19:18:11 -0500
Subject: Use ephemeral socketdir where possible.

At the end of the tests, the agent for testing is still running, and
its sockets are open.

This causes trouble with the debian packaging, because debhelper wants
to install files based on the build tree and gets a bit confused when
there are socket files present.  It will also cause trouble if the
test suite is run on a filesystem that does not support UNIX-domain
sockets (e.g. vfat).

The best way to fix this is probably to ensure that at the end of the
test suite, the agent for the test suite is terminated, e.g. by
running "gpgconf --kill agent" with GNUPGHOME set to tests/testdata as
well.

However, i don't know pytest well enough to know how to get something
to run after all tests are complete.

As a second-best approach, this patch uses gpgconf to create an
ephemeral socket directory outside of the build tree, which gpg-agent
will use.  This isn't very clean, because it should be matched with a
parallel "gpgconf --remove-socketdir" once the agent is terminated.
If i knew where to put that, i'd include it after the "gpgconf --kill
agent".

imho, GnuPG should really be doing this ephemeral socketdir creation
automatically, but that's a separate issue for upstream to deal with,
see https://bugs.gnupg.org/gnupg/issue2964
---
 tests/conftest.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/conftest.py b/tests/conftest.py
index 991833d..14870e3 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -261,6 +261,8 @@ def pytest_configure(config):
     v, _ = _run(_which('pgpdump'), '-v', stderr=subprocess.STDOUT)
     pgpdump_ver.parse(v.split(' ')[2].strip(','))
 
+    _run(_which('gpgconf'), '--create-socketdir', env=_gpg_env)
+    
     # display the working directory and the OpenSSL/GPG/pgpdump versions
     print("Working Directory: " + os.getcwd())
     print("Using OpenSSL " + str(openssl_ver))
